integration/dynamic_namespace_delete_test.go

```diff
@@ -0,0 +1,176 @@
+	require.NoError(t, err)
+	kvStore, err := csClient.KV()
+	require.NoError(t, err)
+
+	// namespace maps
+	protoKey := func(nses ...namespace.Metadata) proto.Message {
+		nsMap, err := namespace.NewMap(nses)
+		require.NoError(t, err)
+		return convert.ToProto(nsMap)
+	}
+
+	// dynamic namespace registry options
+	dynamicOpts := dynamic.NewOptions().
+		SetConfigServiceClient(csClient).
+		SetInstrumentOptions(instrument.NewOptions().SetMetricsScope(scope))
+	dynamicInit := dynamic.NewInitializer(dynamicOpts)
+	testOpts = testOpts.SetNamespaceInitializer(dynamicInit)
+
+	// initialize value in kv
+	_, err = kvStore.Set(dynamicOpts.NamespaceRegistryKey(), protoKey(ns1))
+	require.NoError(t, err)
+
+	// Test setup
+	testSetup, err := newTestSetup(t, testOpts)
+	require.NoError(t, err)
+	defer testSetup.close()
+
+	// Start the server
+	log := testSetup.storageOpts.InstrumentOptions().Logger()
+	require.NoError(t, testSetup.startServer())
+
+	// Stop the server
+	defer func() {
+		require.NoError(t, testSetup.stopServer())
+		log.Infof("server is now down")
+	}()
+
+	// Write test data
+	blockSize := ns0.Options().RetentionOptions().BlockSize()
+	now := testSetup.getNowFn()
+	seriesMaps := make(map[time.Time]generate.SeriesBlock)
+	inputData := []generate.BlockConfig{
+		{[]string{"foo", "bar"}, 100, now},
+		{[]string{"foo", "baz"}, 50, now.Add(blockSize)},
+	}
+	for _, input := range inputData {
+		start := input.Start
+		testData := generate.Block(input)
+		seriesMaps[start] = testData
+	}
+	log.Infof("test data is now generated")
+
+	// fail to write to non-existent namespaces
+	for _, testData := range seriesMaps {
+		require.Error(t, testSetup.writeBatch(ns0.ID(), testData))
+	}
+
+	// delete namespace key, ensure update propagates
+	numInvalid := numInvalidNamespaceUpdates(scope)
+	_, err = kvStore.Delete(dynamicOpts.NamespaceRegistryKey())
+	require.NoError(t, err)
+	log.Infof("deleted namespace key from kv")
+	for {
+		invalidUpdates := numInvalidNamespaceUpdates(scope)
+		if invalidUpdates > numInvalid {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+
+	// update value in kv
+	_, err = kvStore.Set(dynamicOpts.NamespaceRegistryKey(), protoKey(ns0, ns1))
+	require.NoError(t, err)
+	log.Infof("new namespace added to kv")
+
+	// wait until the new namespace is registered
+	nsExists := func() bool {
+		_, ok := testSetup.db.Namespace(ns0.ID())
+		return ok
+	}
+	require.True(t, waitUntil(nsExists, 5*time.Second))
+	log.Infof("new namespace available in testSetup")
+
+	// write to new namespace
+	for start, testData := range seriesMaps {
+		testSetup.setNowFn(start)
+		require.NoError(t, testSetup.writeBatch(ns0.ID(), testData))
+	}
+	log.Infof("test data is now written")
+
+	// Advance time and sleep for a long enough time so data blocks are sealed during ticking
+	testSetup.setNowFn(testSetup.getNowFn().Add(2 * blockSize))
+	later := testSetup.getNowFn()
+	time.Sleep(testOpts.TickInterval() * 10)
```

Comment:
Hm, at some point we should find some way to hint/trigger a manual tick instead of having to sleep for a tick cycle here.

Reply:
I've gone back and forth on that. Doing it this way sucks cause we're just polling & hoping, but it does mimic what the process would be doing in production. If we manually trigger a tick, it'd deviate from production behaviour so we might missing something.

Reply:
Sure thing, I'll make the change to all integration tests where that's applicable.

Reply:
Thanks, big help!

Reply:
sure thing, made the change in the latest patch